common/main/java/com/couchbase/lite/AbstractReplicatorConfiguration.java

```diff
@@ -357,38 +360,43 @@ public final ReplicatorConfiguration setType(@NonNull AbstractReplicator.Replica
     @Nullable
     public final ReplicationFilter getPushFilter() { return pushFilter; }
 
-    /**
-     * Old getter for Replicator type indicating the direction of the replicator.
-     *
-     * @deprecated Use getType()
-     */
-    @Deprecated
-    @NonNull
-    public final ReplicatorType getReplicatorType() {
-        switch (replicatorType) {
-            case PUSH_AND_PULL:
-                return ReplicatorType.PUSH_AND_PULL;
-            case PUSH:
-                return ReplicatorType.PUSH;
-            case PULL:
-                return ReplicatorType.PULL;
-            default:
-                throw new IllegalStateException("Unrecognized replicator type: " + replicatorType);
-        }
-    }
-
     /**
      * Return Replicator type indicating the direction of the replicator.
      */
     @NonNull
-    public final AbstractReplicator.ReplicatorType getType() { return replicatorType; }
+    public final ReplicatorType getReplicatorType() { return replicatorType; }
 
     /**
      * Return the replication target to replicate with.
      */
     @NonNull
     public final Endpoint getTarget() { return target; }
 
+    /**
+     * Return the max number of retry attempts made after connection failure.
+     */
+    final int getMaxRetries() {
+        return (maxRetries >= 0)
+            ? maxRetries
+            : ((continuous)
+                ? AbstractCBLWebSocket.DEFAULT_ONE_SHOT_MAX_RETRIES
+                : AbstractCBLWebSocket.DEFAULT_CONTINUOUS_MAX_RETRIES);
```

Comment:
@pasin Should these remain what they are in 2.8.x until 3.0?  From what I can tell these are the new defaults.

Reply:
I think the changes will be done in 3.0, otherwise java/android will have different behavior from the other platforms.

Reply:
This function is not visible in 2.8.4 (x == 4, unless I am mistaken).  Whether these are the correct values seems a question only for 3.0.0, I think.

Unless you want me to actually remove them.

The amount of time this simple backport has taken is pretty staggering.  I wish we could find a more efficient way of doing this kind of stuff...

Reply:
I might not understand this correctly. It seems like maxRetries is default to -1 so it means that getMaxRetries() will either use DEFAULT_ONE_SHOT_MAX_RETRIES or DEFAULT_CONTINUOUS_MAX_RETRIES?

Reply:
DEFAULT_ONE_SHOT_MAX_RETRIES or DEFAULT_CONTINUOUS_MAX_RETRIES are for Lithium. It's OK to keep getMaxRetries() to use in Lithium but can you remove or comment out line 452 and 453 that use the value?